examples/foo-missionmodel/src/main/java/gov/nasa/jpl/aerie/foomissionmodel/Mission.java

```diff
@@ -74,13 +77,20 @@ public Mission(final Registrar registrar, final Instant planStart, final Configu
     registrar.real("/simple_data/b/rate", this.simpleData.b.rate);
     registrar.real("/simple_data/total_volume", this.simpleData.totalVolume);
 
+    registrar.discrete("/counter", this.counter, new IntegerValueMapper());
+
     spawn(timeTrackerDaemon::run);
 
-    spawn(() -> { // Register a never-ending daemon task
-      while (true) {
-        ModelActions.delay(Duration.SECOND);
+    spawn(replaying(new Runnable() {
+      @Override
+      public void run() { // Register a never-ending daemon task
+        for (int i = 0; i < 1000; i++) {
+          ModelActions.delay(Duration.SECOND);
+        }
+        counter.add(1);
+        spawn(this);
```

Comment:
I worry about this recursive structure on long plans, since the original task waits on L91 for its method to be complete, and the task spawned from that waits for its spawned task, and so on, leading to large stack depths. For scale, if this task was to be run on a 5y plan, it would reach a stack depth of 157,700, which is definitely enough to overflow (frankly, it only takes 1/3 of a year for the depth to reach 10,000). 

Is infinite recursion a recommended demon task structure? If so, does Aerie do anything to reduce the stack size?

Reply:
Hi Theresa, this is a technique pioneered by @DavidLegg that we've been calling "trampolining tasks" - effectively using tail recursion instead of loops to perform repeated tasks. The motivation for these is to make "resuming" a task a quicker operation. A task with in a loop in it has to pay a high resumption cost because its state lives in a paused thread - for tasks that need to resume very many times, the thread switching overhead adds up*. The advantage of "trampolining tasks" is that they are effectively stateless, and they never need to be resumed. When paired with the "replaying task" feature, that means these tasks live entirely in the main simulation thread.

You're right to point out that the cost of this is the memory overhead associated with tracking the task dependencies - I want to avoid using the words "stack depth" or "overflow" here, because I want to be clear that we are not talking about the Java stack, but rather a task decomposition hierarchy. #1174 was intended to address that, and @martyvona has identified some relevant improvements to this issue. Addressing that issue is outside the scope of this PR.

The reason that trampolining tasks are a good fit for duplicatable simulation is that they have very short read histories. In order to duplicate a simulation engine, this PR saves every cell value observed by that task - the fewer of those, the cheaper the duplicate operation is, both in terms of time and memory.

*Side note: when we have time, we should test whether this overhead gets any cheaper with Java 21 - early tests from last year have been inconclusive